github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitEvent.java

```diff
@@ -29,51 +29,51 @@
 import javax.annotation.Nonnull;
 
 import java.io.IOException;
+import java.io.PrintStream;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
 /**
- * This branch event acts as a wrapper around checks that can be performed against commit data that requires an additional round trip to
- * GitHub to retrieve.
+ * This branch event acts as a wrapper around checks that can be performed against commit data that requires
+ * an additional round trip to GitHub to retrieve.
  * <p>
- * Commit data is retrieved and then passed to each implementing instance of <code>GitHubBranchCommitCheck</code> to determine information
- * about the commit should trigger a build.
- * </p>
- * <p>
- * The implementation itself should return a <code>GitHubBranchCause</code> object indicating the build should be skipped, otherwise
- * it should return <code>null</code>.
+ * Commit data is retrieved and then passed to each implementing instance of <code>GitHubBranchCommitCheck</code>
+ * to determine information about the commit should trigger a build.
  * </p>
  *
  * @author Kanstantsin Shautsou
  * @author Jae Gangemi
  */
-public class GitHubBranchCommitChecks extends GitHubBranchEvent {
+public class GitHubBranchCommitEvent extends GitHubBranchEvent {
     private static final String DISPLAY_NAME = "Commit Checks";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubBranchCommitChecks.class);
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubBranchCommitEvent.class);
 
     private List<GitHubBranchCommitCheck> checks = new ArrayList<>();
 
     /**
      * For groovy UI
      */
     @Restricted(value = NoExternalUse.class)
-    public GitHubBranchCommitChecks() {
+    public GitHubBranchCommitEvent() {
     }
 
     @DataBoundConstructor
-    public GitHubBranchCommitChecks(List<GitHubBranchCommitCheck> checks) {
+    public GitHubBranchCommitEvent(List<GitHubBranchCommitCheck> checks) {
         this.checks = checks;
     }
 
     @Override
     public GitHubBranchCause check(GitHubBranchTrigger trigger, GHBranch remoteBranch, @CheckForNull GitHubBranch localBranch,
-            GitHubBranchRepository localRepo, TaskListener listener)
-        throws IOException {
+                                   GitHubBranchRepository localRepo, TaskListener listener)
+            throws IOException {
+        final PrintStream lloger = listener.getLogger();
+
         if (localBranch == null) {
-            LOGGER.debug("First build of branch [%s] detected.", remoteBranch.getName());
+            LOG.debug("First build of branch '%s' detected, skipping check.", remoteBranch.getName());
+            lloger.println("First build of branch '" + remoteBranch.getName() + "' detected, skipping check.");
```

Comment:
i think there might be a bug where the first build actually doesn't fire but i'm not 100% sure of that yet. it could just be related to github throwing an error and the plugin not having any retry yet.

i wouldn't delay for this though.